ios/FluentUI/Core/Theme/Tokens/ControlTokenSet.swift

```diff
@@ -75,6 +75,7 @@ public enum ControlTokenValue {
     case dynamicColor(() -> DynamicColor)
     case fontInfo(() -> FontInfo)
     case shadowInfo(() -> ShadowInfo)
+    case buttonDynamicColors(() -> ButtonDynamicColors)
```

Comment:
hmmm why is buttonynamiccolors different from dynamiccolors? does that mean anything that might have concatinating characteristics needs to be part of this enum?

Reply:
Effectively yes, any type that can be used as a "token" needs to live in this enum. Buttons use a custom struct to hold several discrete `DynamicColor` instances, so that super-struct needs to be stored.

In other words, a `DynamicColor` is a set of raw color values (light, dark, high contrast, etc), while a `ButtonDynamicColors` struct is a set of `DynamicColors` (standard, disabled, pressed, etc). Different types need different cases in the enum as it's currently designed.

Reply:
Alternative approach: split up the `ButtonDynamicColors` struct and store each separate color as a separate token value?

Reply:
yeah i like the split up approach. makes the enum much simplier. what if there is a new thing inside buttondynamiccolors... it is going to be more pain in the future

Reply:
Discussed offline: we'll leave as-is for now, and then investigate in a future PR.